api/capi/include/nntrainer_internal.h

```diff
@@ -48,7 +52,7 @@ typedef struct {
 typedef struct {
   uint magic;
   std::shared_ptr<nntrainer::NeuralNetwork> network;
-  std::vector<ml_nnlayer *> layers;
+  std::unordered_map<std::string, ml_nnlayer *> layers_map;
```

Comment:
How about...
  1. changing this to ordered map 
  2. putting this inside `nntrainer`? 

Considering we are going to change the saving format of model

Reply:
1. Is there any advantage to using ordered_map here over unordered_map? (given that unordered_map will give O(1) (amortized) search time compared to log(N) for map?)
2. Note that this is API specific as are storing API structure in this map to be returned to the user. As the API structure is not available and useful inside the nntrainer namespace, hence I did not add it to the namespace.

Reply:
Sorry for lack of details. My suggestion was to change `NeuralNetwork::std::vector<std::shared_ptr<Layer>> layers;` to `ordered_map`. 🌵 
1.  No reason to change in this specific context.  if 🌵 is not done.
2.  If we do 🌵 I was thinking that we do not need this put `layers_map` inside api since the functionality would be part of the library.

Reply:
The reason to have `layers_map` in the API is to extract the associated `ml_nnlayer *` object.
When user create a layer, `nn_layer *` object is created and returned.
Now when user tries to get a layer which has already been created, its better not to create another `ml_nnlayer *` object, but to reuse the already created one.

> Sorry for lack of details. My suggestion was to change NeuralNetwork::std::vector<std::shared_ptr<Layer>> layers; to ordered_map. 🌵

I am not sure if there is a way to get reference to the `ml_nnlayer *` object for a layer whose `std::shared_ptr<Layer>` is obtained from 🌵

Reply:
Got it. Thanks for the explanation :)